articles/hardware/breakout/analog-io.md

```diff
@@ -0,0 +1,52 @@
+---
+uid: breakout_analog-io
+title: Breakout Board Analog IO
+hardware: true
+device: analog io
+videoCaption: This visualizes analog IO data using Bonsai visualizers. The "BreakoutAnalogOutput" window shows the values that are being used by the <code>BreakoutAnalogOutput</code> operator to update the analog IO port. However, only analog channel 0 is actually updated because it's the only channel configured as output by the <code>BreakoutBoard</code> operator. The "AnalogData" window shows the voltages at the various pins of the analog IO port. They are being read by the the ADCs on the breakout board and produced in Bonsai by the <code>BreakoutAnalogInput</code> operator. This confirms that only analog channel 0 is outputting the sawtooth waveform.
```

Comment:
I don't see the video shown in the markdown file, so I assume it is written in the templates. Either way, watching the video it looks like the sawtooth that is created has a different rate than the sawtooth read in. This is probably just a visual trick because the history length for each graph is different, but there is also what appears to be a delay between what is input and what is output (i.e., phase shift between visualizers). This is not addressed in the text, so I'm not sure if this is again some trick of the visualizers, but if I was watching this video naively I would be a little confused.

Reply:
I'm going to leave this open for jon to address. There might be some software latencies for displaying the input data. I'm not sure the best way to address this.

Reply:
its the history length. it probably does not need to be addressed.